packages/block-editor/src/store/selectors.js

```diff
@@ -1757,13 +1756,18 @@ export const __experimentalGetAllowedBlocks = createSelector(
 	]
 );
 
-const __experimentalGetParsedPatterns = createSelector(
-	( state ) => {
+export const __experimentalGetParsedPattern = createSelector(
+	( state, patternName ) => {
 		const patterns = state.settings.__experimentalBlockPatterns;
-		return map( patterns, ( pattern ) => ( {
+		const pattern = patterns.find( ( { name } ) => name === patternName );
```

Comment:
You already have the pattern from `__experimentalGetAllowedPatterns`. No need to get `state.settings.__experimentalBlockPatterns` again and filter. Am I missing something?

Reply:
The goal is to have a selector function that can be memoized easily. The pattern name as parameter is immutable. If we used the pattern object then it might happen that we provide a modified/reconstructed pattern object that has a different reference.